core/patina_stacktrace/resolve_stacktrace/src/main.rs

```diff
@@ -0,0 +1,216 @@
+
+    let mut pdb_directory = pdb_directory.trim().to_owned();
+    if pdb_directory.is_empty() {
+        // Read from STACKTRACE_PDB_DIR environment variable
+        if let Ok(env_dir) = std::env::var("STACKTRACE_PDB_DIR") {
+            if !env_dir.is_empty() {
+                pdb_directory = env_dir;
+            }
+        } else {
+            return Err("PDB directory not provided or set in STACKTRACE_PDB_DIR".to_string());
+        }
+    }
+
+    if pdb_directory.is_empty() {
+        return Err("PDB directory path cannot be empty".to_string());
+    }
+
+    let pdb_directory = PathBuf::from(pdb_directory);
+
+    println!("Enter stack trace lines (press Enter twice to finish):");
+    let mut stacktrace = vec![];
+    loop {
+        let mut line = String::new();
+        let bytes_read = io::stdin()
+            .read_line(&mut line)
+            .map_err(|e| format!("Failed to read stack trace line from stdin: {}", e))?;
+
+        if bytes_read == 0 {
+            break;
+        }
+
+        let trimmed = line.trim_end();
+        if trimmed.is_empty() {
+            break;
+        }
+
+        stacktrace.push(trimmed.to_string());
+    }
+
+    Ok((pdb_directory, stacktrace))
+}
+
+/// Parse the stack trace text into a list of stack frames, skipping headers.
+fn create_stack_frames(stack_frames: Vec<String>) -> Vec<StackFrame> {
+    stack_frames
+        .iter()
+        .filter_map(|line| {
+            let line = line.trim();
+            if line.is_empty() {
+                return None;
+            }
+
+            // Skip header line
+            if line.contains("Return Address") {
+                return None;
+            }
+
+            create_stack_frame(line)
+        })
+        .collect()
+}
+
+/// Render the resolved stack frames as a formatted table for display.
+fn dump_stack_frames(stack_frames: Vec<StackFrame>) {
+    let mut table = Table::new();
+    table.load_preset(UTF8_FULL).set_content_arrangement(ContentArrangement::DynamicFullWidth).set_header(vec![
+        Cell::new("#").add_attribute(comfy_table::Attribute::Bold),
+        Cell::new("Source Path").add_attribute(comfy_table::Attribute::Bold),
+        Cell::new("Child-SP").add_attribute(comfy_table::Attribute::Bold),
+        Cell::new("Return Address").add_attribute(comfy_table::Attribute::Bold),
+        Cell::new("Call Site").add_attribute(comfy_table::Attribute::Bold),
+    ]);
+
+    for frame in &stack_frames {
+        let source_path = frame.file.as_deref().unwrap_or(frame.error.as_deref().unwrap_or("<unknown>"));
+        let source_path = format!("{} @ {}", source_path, frame.line.unwrap_or(0));
+        let call_site =
+            format!("{}!{}+0x{:X}", frame.module_name, frame.function.as_deref().unwrap_or("<unknown>"), frame.offset);
+
+        table.add_row(vec![
+            frame.frame_number.clone(),
+            source_path.to_string(),
```

Comment:
nit: Is `to_string()` required here?

```suggestion
            source_path,
```